pkg/cluster/scaler/scaler.go

```diff
@@ -12,6 +18,27 @@ import (
 	"github.com/openshift/openshift-azure/pkg/util/azureclient"
 )
 
+// Factory interface is to create new Scaler instances.
+type Factory interface {
+	New(log *logrus.Entry, ssc azureclient.VirtualMachineScaleSetsClient, vmc azureclient.VirtualMachineScaleSetVMsClient, kubeclient kubeclient.Kubeclient, resourceGroup string, ss *compute.VirtualMachineScaleSet) Scaler
+}
+
+type scalerFactory struct{}
+
+// NewFactory create a new Factory instance.
+func NewFactory() Factory {
+	return &scalerFactory{}
+}
+
+// Scaler is an interface that changes the number of VMs in a scaleset until they meet
+// the 'count' provided in each method. Scale function simply checks which direction
+// to scale up or down.
+type Scaler interface {
```

Comment:
It still feels strange to me that we are exposing methods encapsulated in other methods. Why would a caller want to use ScaleUp or ScaleDown if Scale achieves the same thing?